server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -118,8 +119,9 @@ protected Owner getOwnerByKey(String key) {
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<Product> list(@PathParam("owner_key") String ownerKey,
-                              @QueryParam("product") List<String> productIds) {
+    public List<Product> list(
+        @Verify(Owner.class) @PathParam("owner_key") String ownerKey,
+        @Verify(Product.class) @QueryParam("product") List<String> productIds) {
```

Comment:
This is a GET, so you'd need read access on the owner and the product to list the products in an owner, that seems right.